### ISSUE TITLE
Cross validation in TPOT

### ISSUE THREAD

**Issue by rhiever** [rhiever/tpot#184]
After some discussions with @mfeurer, I think it may be important to use k-fold cross validation in TPOT to help avoid overfitting on the internal test set. This is obviously going to slow TPOT down, so I want to validate this assumption first before rolling it out.

Let's implement 5-fold CV in TPOT and compare it on some benchmarks that "regular" TPOT has been benchmarked against. In particular, let's look at:

1) How much does TPOT currently overfit on data sets? i.e., what is the median delta between the internal CV score and the holdout CV score?

2) Does 5-fold CV lead to significantly improved accuracy?

3) How much longer does TPOT take with 5-fold CV?

**Comment by KhaoticMind** [comment]
I was thinking on something on these lines this days. But not exactly 5-fold. Maybe let the user specify how many folds they want (or even if they want no folding at all - just keep train-test as it is).

Implementing it like this would be interesting if we consider #146, or a variation of it, where the user wants to first find the best classifier with no CV (find the best, even if its overfitting). And latter fine tune the parameters of this one classifier with 3, 5 or even 10-fold CV (I've seen kagglers doing 10 fold CV... dunno the real benefit thou).

**Comment by rhiever** [comment]
👍 to the idea of parameterizing the k-fold CV.

**Comment by dashaub** [comment]
While you're at it, consider allowing a parameter for repeated cross validation. That shouldn't be much of a problem. What could be more interesting--although a more difficult implementation--is only performing the repeated cross validation on a few of the candidate models that achieve the highest score. This would prevent the wasted computation on low-accuracy models while allowing discrimination between the top models that might be missed due to the idiosyncratic sampling of one round of CV.

**Comment by rhiever** [comment]
This feature will be in the upcoming 0.5 release.
